### DISCUSSION TITLE
Question about team subscriptions

### DISCUSSION THREAD

**Discussion by strzibny** [pay-rails/pay#550] [General]
Hi,

I am trying to play with Pay and decided to make teams (`Team` model) the owner of a subscription. This works only if the emails are different, but one person (with same email) could create two team subscriptions. When a webhook comes in, the subscription is created with the team that already exists (as an existing owner).

How do you design this with Pay? Do you just leave it with a `User` and then simply work around it when comes to teams?

Thanks a lot for anybody's insight.

**Comment by excid3** [comment]
If you want to create two subscriptions for the same customer, you just call `subscribe` twice on the same Pay::Customer.

**Comment by strzibny** [comment]
Yes, I understood that much. The problem in the `Team` setup was that Paddle will return the same `user_id` for the same email. So on Pay end, the subscription ended up with a different team in the end (the one that created the Pay::Customer first). 

So now I think I have to model it to be a `User` model, right? I would have to name the different subscriptions, which means I have to pass some identifier to Paddle.

Originally I went with `Team` because I though it would be nice that more users could manage the subscription under one company email.

**Reply by excid3** [reply]
Pay uses Paddle's passthrough parameter for associating records. https://github.com/pay-rails/pay/blob/master/lib/pay/paddle.rb#L42-L54

**Reply by strzibny** [reply]
I know that, but notice this line:

https://github.com/pay-rails/pay/blob/master/lib/pay/paddle/subscription.rb#L27

```ruby
def self.sync(subscription_id, object: nil, name: Pay.default_product_name)
        # Passthrough is not return from this API, so we can't use that
        object ||= OpenStruct.new PaddlePay::Subscription::User.list({subscription_id: subscription_id}).try(:first)

        pay_customer = Pay::Customer.find_by(processor: :paddle, processor_id: object.user_id)

        # If passthrough exists (only on webhooks) we can use it to create the Pay::Customer
        if pay_customer.nil? && object.passthrough
          owner = Pay::Paddle.owner_from_passthrough(object.passthrough)
          pay_customer = owner&.set_payment_processor(:paddle, processor_id: object.user_id)
        end
```

If the `Pay::Customer` is found (same email), the passthrough is not used:

```ruby
# line 30
if pay_customer.nil? && object.passthrough
```
and it's not attached to the record in the passthrough.

It works as long as there is always a unique email, because Paddle group it all under same customer `user_id` (from what I can tell).

I would have to change the condition on line 30 but perhaps that would break other Pay workflows.

Initially from the docs I though I can just make subscription model to be something else than `User` which does work, but only as long as the email is unique.

**Reply by excid3** [reply]
I don't quite understand. `user_id` is an integer, not an email. https://developer.paddle.com/api-reference/b3A6MzA3NDQ3MzA-list-users

**Reply by strzibny** [reply]
Yes, but it will be the same ID. Even if it's a new request to subscribe, the webhook will return the same `user_id` because the email was same.

Basically if Paddle just created a new customer with a new ID (but same email), everything would work.

**Reply by excid3** [reply]
If Paddle is mapping it to the same User in their system, we want it to match in Pay.

**Reply by excid3** [reply]
Feel free to fork and try setting it up in a different way and report back. 👍 

I'm pretty sure the passthrough is only available in webhooks (like the comment mentions) so you can't always rely on the `passthrough` to associate with the Customer.

**Reply by strzibny** [reply]
Yes, so basically I was thinking what could I do, but wanted to make sure I am not missing something obvious. Thanks for the quick replies.

**Reply by excid3** [reply]
Let me know how it goes. It's always trickier than it seems because these APIs are pretty poorly written (except Stripe's). 😅
